java/alizer-api/src/main/java/com/redhat/devtools/alizer/api/ComponentRecognizerImpl.java

```diff
@@ -0,0 +1,301 @@
+        if (directoriesWithComponents.isEmpty()) {
+            return false;
+        }
+        for(Path directoryWithComponent: directoriesWithComponents) {
+            if (directoriesWithComponents.equals(directory) ||
+                    isFirstPathParentOfSecond(directoryWithComponent, directory)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private List<Component> getComponentsWithoutConfigFile(List<Path> directories) throws IOException {
+        List<Component> components = new ArrayList<>();
+        for (Path directory: directories) {
+            Component component = detectComponent(directory, "");
+            // only takes component with languages that have no config file
+            // E.g if the directory consists of javascript files but it doesn't contain a package.json, something is wrong
+            // and we do not consider it as an actual component
+            if (component != null
+                && isValidNoConfigComponent(component)) {
+                components.add(component);
+            }
+        }
+        return components;
+    }
+
+    /**
+     * Return true if it is a component with a language that does not have any config file
+     * E.g if the directory (and the component itself) consists of javascript files without a package.json, something is wrong
+     * and we do not consider it as an actual no config component
+     *
+     * @param component component
+     * @return true if it is a valid component, false otherwise
+     */
+    private boolean isValidNoConfigComponent(Component component) {
+        if (component.getLanguages().isEmpty()) {
+            return false;
+        }
+        String language = component.getLanguages().get(0).getName();
+        LanguageFileItem languageFileItem = LanguageFileHandler.get().getLanguageByNameOrAlias(language);
+        return languageFileItem.getConfigurationFiles().isEmpty();
+    }
+
+    private List<Component> detectComponents(List<File> files) throws IOException {
+        Map<String, String> configurationPerLanguage = LanguageFileHandler.get().getConfigurationPerLanguageMapping();
+        List<Component> components = new ArrayList<>();
+        for (File file: files) {
+            if (configurationPerLanguage.containsKey(file.getName())
+                    && isConfigurationValid(file.toPath(), configurationPerLanguage.get(file.getName()))) {
+                Component component = detectComponent(file.getParentFile().toPath(), configurationPerLanguage.get(file.getName()));
+                if (component != null) {
+                    components.add(component);
+                }
+            }
+        }
+        return components;
+    }
+
+    private boolean isConfigurationValid(Path file, String language) {
+        if (!isValidPathPerLanguage(file, language)) {
+            return false;
+        }
+        return !isParentModuleMaven(file);
```

Comment:
Why do we have this specific ? I thought framework specific stuff was located into SPI